File: lua/hello-world/hello-world.lua
local hello_world = {}

function hello_world.hello (name)
  local greet = "Hello,"
  if name == nil then
    greet = string.format("%s world!", greet) 
  else
    greet = string.format("%s %s!", greet, name)
  end
  return greet
end

return hello_world

File: go/hello-world/hello_world.go
/*
hello-word from exercism in go
*/
package greeting

import (
	"fmt"
)

const testVersion = 3

// Returns "Hello, 'name'!" for every value of name.
// If name is "" then it returns "Hello, World!"
func HelloWorld(name string) string {
	switch name {
	case "":
		return "Hello, World!"
	default:
		return fmt.Sprintf("Hello, %s!", name)
	}
}
